cddl/concise-mid-tag.cddl

```diff
@@ -139,97 +151,113 @@ $device-id-type /= tagged-ueid
 
-; NOTE(tho) from a syntactic PoV reference-value and endorsed-value are extremely similar.  I'm wondering why do we need to keep them separate?
-; NOTE(henk) I very much hope that they are similar: endorsements are on the "evidence side" of things as they kick in when the Attester cannot create evidence about the characteristics of its own components (roots of trust). Reference-values are the things that you compare the "evidence side" with. So endorsements cannot be part of the "reference side". They have to look similar/identical though - otherwise the comparison becomes complex and brittle.
-
-instance-claim-entry = {
-  ? element-name,
-  ; TLVs where instance-type identifies the data type e.g. MAC address, serial# etc.
-  instance-type => instance-values / [2* instance-values],
+instance-value = {
+  ? 0 => element-name
+  $$instance-claim-choice
 }
 
-instance-type /= mac-address
-instance-type /= serial-number
-instance-type /= euid
-instance-type /= uuid
-instance-type /= ip-addr
-
-full-address = [[+ label], ip4, ip6]
-ip4 = bstr .size 4
-ip6 = bstr .size 16
-label = bstr .size (1..63)
-
-mac-address = 0
-serial-number = 1
-euid = 2
-uuid = 3
-ip-addr = 4
-
-instance-values /= full-address
-instance-values /= uuid-bytes
-instance-values /= #6.47115(ueid) ;should define ueid-bytes type?
-instance-values /= mac-address-bytes
-instance-values /= serial-number-bytes
-; ...
+$$instance-claim-choice = (
+  mac-addr-key => mac-addr-type //
+  ip-addrs-key => one-or-many<ip-addr-type> //
+  serial-number-key => serial-number-type //
+  ueid-key => ueid //
+  uuid-key => uuid
+)
+
+mac-addr-key = 1
+ip-addrs-key = 2
+serial-number-key = 3
+ueid-key = 4
+uuid-key = 5
+
+ip-addr-type = ip4-addr-type / ip6-addr-type
+ip4-addr-type = bytes .size 4
+ip6-addr-type = bytes .size 16
+
+mac-addr-type = bytes .size 6
+
+serial-number-type = text
+
 ; Note: are there other well defined types under #6.4711 or EAT etc...
 
-element-name = (
-    ? label => text,
-    ? vendor => text, ; namespace authority designation
-    ? type => $module-type, ; Note: 'type' SHOULD include a namespace authority if 'vendor' is ommitted
-    ? model => text,
-    ? layer => int,
-    ? index => int,
-; **note: An extension point in element-name confuses verifiers' ability to match against evidence**
-)
-; label is defined in coswid schema (q: should we use a different value if the semantics differ?)
-; type is defined in coswid schema
-vendor=110
-model=111
-layer=112
-index=113
-
-element-value = (
+; Notes:
+; - vendor is the namespace authority designation
+; - type SHOULD include a namespace authority if vendor is ommitted
+; - An extension point in element-name confuses verifiers' ability to match
+;   against evidence
+; - label is defined in coswid schema (q: should we use a different value if
+;   the semantics differ?)
+; - type is defined in coswid schema
+element-name = {
+  ? label => text,
+  ? vendor => text,
```

Comment:
From the top of the head: is this (e.g. vendor) not a conflict with another cddl fragment?